web/client/plugins/DynamicLegend.jsx

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable react/jsx-boolean-value */
+import React from 'react';
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+import { get } from 'lodash';
+import { Glyphicon } from 'react-bootstrap';
+
+import { createPlugin } from '../utils/PluginsUtils';
+import { groupsSelector, layersSelector } from '../selectors/layers';
+import { keepLayer } from '../selectors/dynamiclegend';
+import { mapSelector } from '../selectors/map';
+import { updateNode } from '../actions/layers';
+import controls from '../reducers/controls';
+import { toggleControl } from '../actions/controls';
+import Message from '../components/I18N/Message';
+
+import DynamicLegend from './dynamicLegend/components/DynamicLegend';
+
+export default createPlugin('DynamicLegend', {
+    component: connect(
+        createSelector([
+            (state) => get(state, 'controls.dynamic-legend.enabled'),
+            groupsSelector,
+            layersSelector,
+            mapSelector
+        ], (isVisible, groups, layers, map) => ({
+            isVisible,
+            groups,
+            layers: layers.filter(keepLayer),
+            currentZoomLvl: map?.zoom,
+            mapBbox: map?.bbox
+        })),
+        {
+            onClose: toggleControl.bind(null, 'dynamic-legend', null),
+            onUpdateNode: updateNode
+        }
+    )(DynamicLegend),
+    options: {
+        disablePluginIf: "{state('router') && (state('router').endsWith('new') || state('router').includes('newgeostory') || state('router').endsWith('dashboard'))}"
+    },

```

Comment:
In In https://github.com/allyoucanmap/MapStore2/commit/a11b6736dd0cc5175977ec75a212d3c7d205d309 removed because not needed